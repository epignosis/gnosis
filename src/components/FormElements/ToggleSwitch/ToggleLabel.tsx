import React from "react";
import classNames from "classnames";
import { Text, Tooltip } from "../../../";
import { InfoIconSVG } from "../../../icons";

type ToggleLabelProps = {
  label: string;
  position: "before" | "after";
  description?: string;
  tooltip?: string;
  required?: boolean;
  notSwitchedOff?: boolean;
  hasDescription?: boolean;
  onClick: (e: React.MouseEvent<HTMLDivElement>) => void;
};

const labelClassNames = (
  customClassName: string,
  notSwitchedOff: boolean,
  isRequired: boolean,
): string =>
  classNames(customClassName, {
    "binary-bold": notSwitchedOff,
    required: isRequired,
  });

const ToggleLabel: React.FC<ToggleLabelProps> = ({
  label,
  position,
  description,
  tooltip,
  required = false,
  notSwitchedOff = false,
  hasDescription = false,
  onClick,
}) => {
  const hasDescriptionTextWeight = hasDescription ? "700" : "400";
  const isBeforeLabel = position === "before";

  return (
    <div className={`label-description-container label-${position}`}>
      <Text
        fontSize="sm"
        className={labelClassNames(
          `label ${isBeforeLabel ? "is-before" : ""}`,
          notSwitchedOff,
          required,
        )}
        onClick={onClick}
        as="div"
        weight={hasDescriptionTextWeight}
      >
        {tooltip && isBeforeLabel && (
          <Tooltip content={tooltip} parentProps={{ "aria-label": tooltip }}>
            <InfoIconSVG height={16} />
          </Tooltip>
        )}
        {label}
        {tooltip && !isBeforeLabel && (
          <Tooltip content={tooltip} parentProps={{ "aria-label": tooltip }}>
            <InfoIconSVG height={16} />
          </Tooltip>
        )}
      </Text>
      {description && (
        <Text fontSize="sm" as="div" className="description-text">
          {description}
        </Text>
      )}
    </div>
  );
};

export default ToggleLabel;
