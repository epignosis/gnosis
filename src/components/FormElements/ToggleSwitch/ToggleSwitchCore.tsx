import React from "react";
import classNames from "classnames";
import { Text } from "../../../";

export type ToggleSwitchVariant = "primary" | "success";
export type ToggleSwitchSize = "sm" | "md";

type ToggleSwitchCoreProps = {
  isChecked: boolean;
  variant: ToggleSwitchVariant;
  size: ToggleSwitchSize;
  hasInlineText?: boolean;
  inlineTextTranslations?: {
    enabled: string;
    disabled: string;
  };
  InternalIcon?: JSX.Element;
  labelledById?: string;
  onClick: (e: React.MouseEvent<HTMLDivElement>) => void;
};

const switchClassNames = (customClassName: string, isMedium: boolean, isSuccess: boolean): string =>
  classNames(customClassName, {
    md: isMedium,
    success: isSuccess,
  });

const ToggleSwitchCore: React.FC<ToggleSwitchCoreProps> = ({
  isChecked,
  variant,
  size,
  hasInlineText = false,
  inlineTextTranslations = { enabled: "Enabled", disabled: "Disabled" },
  InternalIcon,
  labelledById,
  onClick,
}) => {
  const isMedium = size === "md";
  const isSuccess = variant === "success";
  const hasIcon = Boolean(InternalIcon);

  return (
    <div className="switch-wrapper">
      <div
        data-testid="switch"
        data-checked={isChecked}
        className={switchClassNames("switch", isMedium, isSuccess)}
        onClick={onClick}
        role="switch"
        aria-checked={isChecked}
        aria-labelledby={labelledById}
      >
        {hasInlineText && isMedium && (
          <Text fontSize="sm" className="inline-text">
            {isChecked ? inlineTextTranslations.enabled : inlineTextTranslations.disabled}
          </Text>
        )}

        {hasIcon && <div className="internal-icon">{InternalIcon}</div>}
        <div className={switchClassNames("thumb", isMedium, isSuccess)} />
      </div>
    </div>
  );
};

export default ToggleSwitchCore;
