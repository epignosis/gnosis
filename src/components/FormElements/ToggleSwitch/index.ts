export { default as ToggleSwitch } from "./ToggleSwitch";
export { default as ToggleLabel } from "./ToggleLabel";
export { default as ToggleSwitchCore } from "./ToggleSwitchCore";
export type { ToggleProps, ToggleSwitchHandlers } from "./ToggleSwitch";
export type { ToggleSwitchVariant, ToggleSwitchSize } from "./ToggleSwitchCore";
export type { ToggleStyleVariant, ToggleStyleSize } from "./styles";
