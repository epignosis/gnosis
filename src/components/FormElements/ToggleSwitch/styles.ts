import { css, SerializedStyles, Theme } from "@emotion/react";

export const ToggleContainer = (
  { formElements }: Theme,
  {
    isChecked,
    isDisabled,
    hasDescription,
    notSwitchedOff,
  }: { isChecked: boolean; isDisabled: boolean; hasDescription: boolean; notSwitchedOff: boolean },
): SerializedStyles => css`
  display: flex;
  flex-direction: column;
  gap: 5px;

  .switch-container {
    display: inline-block;
    cursor: ${isDisabled ? "not-allowed" : "pointer"};
    display: flex;
    justify-content: flex-start;
    align-items: ${hasDescription ? "flex-start" : "center"};
    width: fit-content;

    .switch-container {
      display: flex;

      .switch {
        position: relative;
        display: inline-block;
        width: 33px;
        height: 17px;
        background-color: ${isDisabled
          ? formElements.toggleSwitch.disabledBackground
          : isChecked
          ? formElements.toggleSwitch.backgroundEnabled
          : notSwitchedOff
          ? formElements.toggleSwitch.backgroundEnabled
          : formElements.toggleSwitch.backgroundDisabled};
        border-radius: 10px;
        transition: background-color 0.3s;
        margin-top: ${hasDescription ? "0.25rem" : "0"};

        &.md {
          width: 109px;
          height: 40px;
          border-radius: 32px;
        }

        &.success {
          background-color: ${isDisabled
            ? formElements.toggleSwitch.disabledBackground
            : isChecked
            ? formElements.toggleSwitch.backgroundInlineEnabled
            : formElements.toggleSwitch.backgroundInlineDisabled};

          &:hover {
            background-color: ${isChecked
              ? formElements.toggleSwitch.backgroundInlineEnabled
              : formElements.toggleSwitch.hoverDisabledColor};
          }
        }
      }

      .thumb {
        position: absolute;
        top: 1px;
        margin-inline-start: ${isChecked ? "16px" : "2px"};
        width: 15px;
        height: 15px;
        background-color: ${formElements.toggleSwitch.textColor};
        border-radius: 50%;
        transition: 0.3s ease-in-out;

        &.md {
          top: 4px;
          border-radius: 32px;
          margin-inline-start: ${isChecked ? "70px" : "5px"};
          width: 32px;
          height: 32px;
        }
      }

      .inline-text {
        position: absolute;
        top: 9px;
        margin-inline-start: ${isChecked ? "12px" : "42px"};
        color: ${formElements.toggleSwitch.textColor};
      }
    }

    .label-description-container {
      display: flex;
      flex-direction: column;

      &.label-before {
        margin-inline-end: 0.5rem;
      }

      &.label-after {
        margin-inline-start: 0.5rem;
      }

      .label {
        margin-inline-start: unset;
        cursor: pointer;
        white-space: nowrap;
        display: flex;
        align-items: center;
        gap: 0.5em;

        &.binary-bold {
          cursor: default;

          &.is-before {
            font-weight: ${isChecked ? 400 : 700};
          }

          font-weight: ${isChecked ? 700 : 400};
        }

        &.required {
          &::after {
            content: " *";
            color: ${formElements.toggleSwitch.requiredAsteriskColor};
          }
        }

        & > div {
          line-height: 0;

          svg {
            color: ${formElements.toggleSwitch.backgroundEnabled};
          }
        }
      }
    }
  }

  .subtitle-text {
    color: ${formElements.toggleSwitch.backgroundDisabled};
  }
`;
