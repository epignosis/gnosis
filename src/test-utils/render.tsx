import React, { FC, ReactElement, ComponentType } from "react";
import { render, RenderResult, RenderOptions } from "@testing-library/react";
import ThemeProvider from "../components/ThemeProvider/ThemeProvider";

const AllTheProviders: FC = ({ children }) => <ThemeProvider>{children}</ThemeProvider>;

const customRender = (ui: ReactElement, options: RenderOptions): RenderResult =>
  render(ui, { wrapper: AllTheProviders as ComponentType, ...options });

export * from "@testing-library/react";

export { customRender as render };
