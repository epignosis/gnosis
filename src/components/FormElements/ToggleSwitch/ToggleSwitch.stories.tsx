import React from "react";
import { StoryFn } from "@storybook/react";
import { PreviewIconSVG } from "../../../icons";
import ToggleSwitch, { ToggleProps } from "./ToggleSwitch";

export default {
  title: "components/Form Elements/ToggleSwitch",
  component: ToggleSwitch,
  args: {
    id: "toggle-switch",
    labelBefore: "Before",
    inlineTextTranslations: { enabled: "Enabled", disabled: "Disabled" },
  },
  decorators: [
    (Story: StoryFn): JSX.Element => (
      <div style={{ maxWidth: 500 }}>
        <Story />
      </div>
    ),
  ],
};

const Template: StoryFn<ToggleProps> = (args) => <ToggleSwitch {...args} />;

export const Default = Template.bind({});

export const WithInternalIcon = Template.bind({});

WithInternalIcon.args = {
  size: "md",
  InternalIcon: <PreviewIconSVG height={32} />,
};
