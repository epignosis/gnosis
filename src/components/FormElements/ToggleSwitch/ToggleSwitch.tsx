import React, { useState, useEffect } from "react";
import { SerializedStyles } from "@emotion/utils";
import classNames from "classnames";
import { Text, Tooltip } from "../../../";
import { InfoIconSVG } from "../../../icons";
import { ToggleContainer } from "./styles";

export type ToggleProps = {
  id?: string;
  labelBefore?: string;
  labelAfter?: string;
  defaultChecked?: boolean;
  description?: string;
  isDisabled?: boolean;
  required?: boolean;
  tooltip?: string;
  variant?: "primary" | "success";
  size?: "sm" | "md";
  notSwitchedOff?: boolean;
  subtitle?: string;
  hasInlineText?: boolean;
  inlineTextTranslations?: {
    enabled: string;
    disabled: string;
  };
  [key: string]: unknown;
  InternalIcon?: JSX.Element;
  onChange?: () => void;
};

const labelClassNames = (
  customClassName: string,
  notSwitchedOff: boolean,
  isRequired: boolean,
): string =>
  classNames(customClassName, {
    "binary-bold": notSwitchedOff,
    required: isRequired,
  });

const switchClassNames = (customClassName: string, isMedium: boolean, isSuccess: boolean): string =>
  classNames(customClassName, {
    md: isMedium,
    success: isSuccess,
  });

const ToggleSwitch: React.FC<ToggleProps> = ({
  id = "toggle-switch",
  labelBefore,
  labelAfter,
  description,
  defaultChecked = false,
  isDisabled = false,
  required = false,
  hasInlineText = false,
  inlineTextTranslations = { enabled: "Enabled", disabled: "Disabled" },
  variant = "primary",
  notSwitchedOff = false,
  size = "sm",
  subtitle = "",
  tooltip,
  InternalIcon,
  onChange,
  ...rest
}) => {
  const [isChecked, setIsChecked] = useState(defaultChecked);
  const hasDescription = Boolean(description);
  const hasDescriptionTextWeight = hasDescription ? "700" : "400";
  const isMedium = size === "md";
  const isSuccess = variant === "success";
  const hasIcon = Boolean(InternalIcon);

  useEffect(() => {
    setIsChecked(defaultChecked);
  }, [defaultChecked]);

  const handleToggle = (e: React.MouseEvent<HTMLDivElement>): void => {
    e.stopPropagation();

    if (!isDisabled) {
      setIsChecked((prev) => !prev);
      if (onChange) onChange();
    }
  };

  return (
    <div
      css={(theme): SerializedStyles =>
        ToggleContainer(theme, {
          isChecked,
          isDisabled,
          hasDescription,
          notSwitchedOff,
        })
      }
      data-testid={id}
      data-checked={isChecked}
      {...rest}
    >
      <div className="switch-container">
        {labelBefore && (
          <div className="label-description-container label-before">
            <Text
              fontSize="sm"
              className={labelClassNames("label is-before", notSwitchedOff, false)}
              onClick={handleToggle}
              weight={hasDescriptionTextWeight}
            >
              {tooltip && (
                <Tooltip content={tooltip} parentProps={{ "aria-label": tooltip }}>
                  <InfoIconSVG height={16} />
                </Tooltip>
              )}
              {labelBefore}
            </Text>
            <Text fontSize="sm" as="div" className="description-text">
              {description}
            </Text>
          </div>
        )}

        <div className="switch-container">
          <div
            data-testid="switch"
            data-checked={isChecked}
            className={switchClassNames("switch", isMedium, isSuccess)}
            onClick={handleToggle}
          >
            {hasInlineText && isMedium && (
              <Text fontSize="sm" className="inline-text">
                {isChecked ? inlineTextTranslations.enabled : inlineTextTranslations.disabled}
              </Text>
            )}

            {hasIcon && <div className="internal-icon">{InternalIcon}</div>}
            <div className={switchClassNames("thumb", isMedium, isSuccess)} />
          </div>
        </div>

        {labelAfter && (
          <div className="label-description-container label-after">
            <Text
              fontSize="sm"
              className={labelClassNames("label", notSwitchedOff, required)}
              onClick={handleToggle}
              as="div"
              weight={hasDescriptionTextWeight}
            >
              {labelAfter}
              {tooltip && (
                <Tooltip content={tooltip} parentProps={{ "aria-label": tooltip }}>
                  <InfoIconSVG height={16} />
                </Tooltip>
              )}
            </Text>
            {description && (
              <Text fontSize="sm" as="div" className="description-text">
                {description}
              </Text>
            )}
          </div>
        )}
      </div>
      {subtitle && (
        <div className="subtitle-text">
          <Text fontSize={"sm"}>{subtitle}</Text>
        </div>
      )}
    </div>
  );
};

export default ToggleSwitch;
