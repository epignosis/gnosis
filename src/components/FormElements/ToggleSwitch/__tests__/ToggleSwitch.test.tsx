import React, { createRef } from "react";
import { act } from "@testing-library/react";
import ToggleSwitch, { type ToggleSwitchHandlers } from "../ToggleSwitch";
import { render, fireEvent, screen } from "@test-utils/render";

describe("<ToggleSwitch />", () => {
  const defaultProps = {
    id: "test-toggle",
  };

  describe("Basic Functionality", () => {
    it("renders with correct default state", () => {
      render(<ToggleSwitch {...defaultProps} />);

      const toggle = screen.getByTestId("switch");
      const container = screen.getByTestId("test-toggle");

      expect(toggle).toHaveAttribute("data-checked", "false");
      expect(container).toHaveAttribute("data-checked", "false");
      expect(toggle).toHaveAttribute("role", "switch");
      expect(toggle).toHaveAttribute("aria-checked", "false");
    });

    it("renders with defaultChecked=true", () => {
      render(<ToggleSwitch {...defaultProps} defaultChecked={true} />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toHaveAttribute("data-checked", "true");
      expect(toggle).toHaveAttribute("aria-checked", "true");
    });

    it("toggles state when clicked", () => {
      render(<ToggleSwitch {...defaultProps} />);

      const toggle = screen.getByTestId("switch");

      expect(toggle).toHaveAttribute("data-checked", "false");

      fireEvent.click(toggle);
      expect(toggle).toHaveAttribute("data-checked", "true");

      fireEvent.click(toggle);
      expect(toggle).toHaveAttribute("data-checked", "false");
    });

    it("toggles state when Enter key is pressed", () => {
      render(<ToggleSwitch {...defaultProps} />);

      const container = screen.getByTestId("test-toggle");
      const toggle = screen.getByTestId("switch");

      expect(toggle).toHaveAttribute("data-checked", "false");

      fireEvent.keyDown(container, { key: "Enter" });
      expect(toggle).toHaveAttribute("data-checked", "true");
    });

    it("calls onChange callback with correct value", () => {
      const mockOnChange = jest.fn();
      render(<ToggleSwitch {...defaultProps} onChange={mockOnChange} />);

      const toggle = screen.getByTestId("switch");

      fireEvent.click(toggle);
      expect(mockOnChange).toHaveBeenCalledWith(true);

      fireEvent.click(toggle);
      expect(mockOnChange).toHaveBeenCalledWith(false);
    });

    it("does not toggle when disabled", () => {
      const mockOnChange = jest.fn();
      render(<ToggleSwitch {...defaultProps} isDisabled={true} onChange={mockOnChange} />);

      const toggle = screen.getByTestId("switch");

      fireEvent.click(toggle);
      expect(toggle).toHaveAttribute("data-checked", "false");
      expect(mockOnChange).not.toHaveBeenCalled();
    });

    it("does not change state when preventToggle is true", () => {
      const mockOnChange = jest.fn();
      render(<ToggleSwitch {...defaultProps} preventToggle={true} onChange={mockOnChange} />);

      const toggle = screen.getByTestId("switch");

      fireEvent.click(toggle);
      expect(toggle).toHaveAttribute("data-checked", "false");
      expect(mockOnChange).toHaveBeenCalledWith(false);
    });
  });

  describe("Labels and Descriptions", () => {
    it("renders labelBefore correctly", () => {
      render(
        <ToggleSwitch
          {...defaultProps}
          labelBefore="Enable feature"
          description="This enables the feature"
        />,
      );

      expect(screen.getByText("Enable feature")).toBeInTheDocument();
      expect(screen.getByText("This enables the feature")).toBeInTheDocument();
    });

    it("renders labelAfter correctly", () => {
      render(<ToggleSwitch {...defaultProps} labelAfter="Feature enabled" required={true} />);

      expect(screen.getByText("Feature enabled")).toBeInTheDocument();
    });

    it("renders subtitle correctly", () => {
      render(<ToggleSwitch {...defaultProps} subtitle="Additional info" />);

      expect(screen.getByText("Additional info")).toBeInTheDocument();
    });

    it("renders tooltip when provided", () => {
      render(
        <ToggleSwitch
          {...defaultProps}
          labelBefore="Toggle"
          tooltip="This is helpful information"
        />,
      );

      const container = screen.getByTestId("test-toggle");
      expect(container).toHaveAttribute("aria-label", "This is helpful information");
    });
  });

  describe("Variants and Sizes", () => {
    it("applies primary variant correctly", () => {
      render(<ToggleSwitch {...defaultProps} variant="primary" />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toBeInTheDocument();
    });

    it("applies success variant correctly", () => {
      render(<ToggleSwitch {...defaultProps} variant="success" />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toHaveClass("success");
    });

    it("applies small size correctly", () => {
      render(<ToggleSwitch {...defaultProps} size="sm" />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).not.toHaveClass("md");
    });

    it("applies medium size correctly", () => {
      render(<ToggleSwitch {...defaultProps} size="md" />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toHaveClass("md");
    });

    it("renders inline text for medium size when enabled", () => {
      render(
        <ToggleSwitch
          {...defaultProps}
          size="md"
          hasInlineText={true}
          defaultChecked={true}
          inlineTextTranslations={{ enabled: "ON", disabled: "OFF" }}
        />,
      );

      expect(screen.getByText("ON")).toBeInTheDocument();
    });

    it("renders internal icon when provided", () => {
      const TestIcon = <span data-testid="test-icon">✓</span>;
      render(<ToggleSwitch {...defaultProps} InternalIcon={TestIcon} />);

      expect(screen.getByTestId("test-icon")).toBeInTheDocument();
    });
  });

  describe("Accessibility", () => {
    it("has proper ARIA attributes", () => {
      render(<ToggleSwitch {...defaultProps} labelledById="external-label" />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toHaveAttribute("role", "switch");
      expect(toggle).toHaveAttribute("aria-checked", "false");
      expect(toggle).toHaveAttribute("aria-labelledby", "external-label");
    });

    it("is keyboard accessible", () => {
      render(<ToggleSwitch {...defaultProps} />);

      const container = screen.getByTestId("test-toggle");
      expect(container).toHaveAttribute("tabIndex", "0");
    });
  });

  describe("Imperative API", () => {
    it("exposes toggle method via ref", () => {
      const ref = createRef<ToggleSwitchHandlers>();
      render(<ToggleSwitch {...defaultProps} ref={ref} />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toHaveAttribute("data-checked", "false");

      act(() => {
        ref.current?.toggle();
      });
      expect(toggle).toHaveAttribute("data-checked", "true");

      act(() => {
        ref.current?.toggle();
      });
      expect(toggle).toHaveAttribute("data-checked", "false");
    });
  });

  describe("Edge Cases", () => {
    it("handles missing inlineTextTranslations gracefully", () => {
      render(
        <ToggleSwitch {...defaultProps} size="md" hasInlineText={true} defaultChecked={true} />,
      );

      expect(screen.getByText("Enabled")).toBeInTheDocument();
    });

    it("stops event propagation on click", () => {
      const mockParentClick = jest.fn();
      render(
        <div onClick={mockParentClick}>
          <ToggleSwitch {...defaultProps} />
        </div>,
      );

      const toggle = screen.getByTestId("switch");
      fireEvent.click(toggle);

      expect(mockParentClick).not.toHaveBeenCalled();
    });

    it("updates state when defaultChecked prop changes", () => {
      const { rerender } = render(<ToggleSwitch {...defaultProps} defaultChecked={false} />);

      const toggle = screen.getByTestId("switch");
      expect(toggle).toHaveAttribute("data-checked", "false");

      rerender(<ToggleSwitch {...defaultProps} defaultChecked={true} />);
      expect(toggle).toHaveAttribute("data-checked", "true");
    });
  });

  describe("Snapshot", () => {
    it("matches snapshot with all props", () => {
      const { container } = render(
        <ToggleSwitch
          id="snapshot-toggle"
          labelBefore="Before Label"
          labelAfter="After Label"
          description="Test description"
          subtitle="Test subtitle"
          variant="success"
          size="md"
          hasInlineText={true}
          required={true}
          tooltip="Test tooltip"
          defaultChecked={true}
          showOutlineBorder={true}
          InternalIcon={<span>✓</span>}
        />,
      );

      expect(container).toMatchSnapshot();
    });
  });
});
