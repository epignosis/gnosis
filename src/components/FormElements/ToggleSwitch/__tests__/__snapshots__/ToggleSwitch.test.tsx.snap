// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> Matches snapshot 1`] = `
<div>
  <div
    class="css-m7865s-ToggleContainer-ToggleContainer"
    data-checked="false"
    data-testid="my-toggle"
    id="my-toggle"
    tabindex="0"
  >
    <div
      class="switch-container"
    >
      <div
        class="switch-container"
      >
        <div
          aria-checked="false"
          aria-labelledby="text-id"
          class="switch"
          data-checked="false"
          data-testid="switch"
          role="switch"
        >
          <div
            class="thumb"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
